src/fmtargs.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Redis Contributors.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD 3-Clause
+ *
+ * To make it easier to map each part of the format string with each argument,
+ * this file provides a way to write
+ *
+ *     printf("a = %s, b = %s, c= %s\n",
+ *            arg1, arg2, arg3);
+ *
+ * as
+ *
+ *     printf(FMTARGS("a = %s, ", arg1,
+ *                    "b = %s, ", arg2,
+ *                    "c = %s\n", arg3));
```

Comment:
i think we can include here a line or two about how it is done (i.e. detecting how many arguments there are and re-ordering them)

Reply:
I think it's tricky to explain how it works in a line or two. :)

Here's a try in 5 lines:

> It is variadic macro which is implemented by passing on `__VA_ARGS__` to two
> other variadic macros of which one of them extracts the formats and the other
> extracts the arguments. The definitions of these macros include counting the
> number of arguments. Therefore, they don't accept an unlimited number of
> arguments. Currently it is fixed to a maximum of 120 formats and arguments.

Reply:
that text seems good..
i think that any explanation, we add can help the reader.
for us it may be easier to understand the code since we already know what it does and how it works, but for a new reader, any trivial piece of info can be gold.